crates/ra_parser/src/grammar/items.rs

```diff
@@ -187,6 +195,9 @@ pub(super) fn maybe_item(p: &mut Parser, m: Marker, flavor: ItemFlavor) -> Resul
         // test default_impl
         // default impl Foo {}
 
+        // test default_unsafe_impl
+        // default unsafe impl Foo {}
```

Comment:
This test should move moved to the `if` condition where we test for `default unsafe`.